### PR TITLE
Setting up test infrastructure using rewire

### PR DESCRIPTION
**Pull request by srmukher**
Setting up test infrastructure using rewire.

### CODE REVIEW

File: tools/UnitTestApp/Sparkline/SparklineUT.test.tsx [hunk 1]
```diff
@@ -43,21 +44,19 @@ const sparkline1Points = {
...
-    expect(result).toBe(false);
-  });
-  test('Should return true when the chart is empty', () => {
-    const rewire = require('rewire');
-    const SparklineBaseModule = rewire('./SparklineBase.js');
-    const SparklineBase = SparklineBaseModule.__get__('SparklineBase');
-    const instance = new SparklineBase({data: [], width: 100, height: 100, _valueTextWidth: 100});
-    const result = instance._isChartEmpty();
-    expect(result).toBe(true);
+const pathToBaseModule = 'D:/fluentui_contrib/fluentui-charting-contrib/tools/UnitTestApp/node_modules/@fluentui/react-charting/lib-commonjs/components/Sparkline/Sparkline.base.js';
```
Comment (AtishayMsft) [inline tools/UnitTestApp/Sparkline/SparklineUT.test.tsx:47]: path should become relative

File: tools/UnitTestApp/TestUtility.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+function testWithRewire(pathToModule, baseClass, testFunction) {
+    const rewire = require('rewire');
+    const module = rewire(pathToModule);
+    const AreaChartBase = module.__get__(baseClass);
```
Comment (AtishayMsft) [inline tools/UnitTestApp/TestUtility.ts:4]: why is it called AreaChartBase

File: tools/UnitTestApp/script.ts [hunk 3]
```diff
@@ -1,24 +1,91 @@
...
+const testSetupRewire = () => {
+  readDirectory('D:/fluentui_contrib/fluentui-charting-contrib/tools/UnitTestApp/node_modules/@fluentui/react-charting/lib-commonjs/');
+  readDirectoryDeep('D:/fluentui_contrib/fluentui-charting-contrib/tools/UnitTestApp/node_modules/@fluentui/react-charting/lib-commonjs/components/');
+  const indexFilePath = 'D:/fluentui_contrib/fluentui-charting-contrib/tools/UnitTestApp/node_modules/@fluentui/react-charting/lib-commonjs/index.js';
+  setupIndexFile(indexFilePath, null);
 };
 
-updatedFile();
\ No newline at end of file
+testSetupRewire();
```
Comment (AtishayMsft) [inline tools/UnitTestApp/script.ts:90]: does this test function need to be exported

File: tools/UnitTestApp/script.ts [hunk 4]
```diff
@@ -1,24 +1,91 @@
...
-      result = result.replace(/require\("d3-array"\)/g, "import('d3-array')");
-      // Write the file back
-      fs.writeFile(writeFilePath, result, 'utf8', err => {
-        if (err) {
-          console.error(`Error writing file to disk: ${err}`);
-        }
-      });
+async function updatedFileDeep(filePath) {
+  let data = await fs.readFileSync(filePath, 'utf8'); 
+  let matches = data.match(/.*=.*d3.*(\n.*?)*(?=var|$)/g);
```
Comment (AtishayMsft) [inline tools/UnitTestApp/script.ts:22]: add a comment about what this regex denote

File: tools/UnitTestApp/script.ts [hunk 5]
```diff
@@ -1,24 +1,91 @@
...
+      }
+    });
+    // Write the file back
+    if (data != null && data != '' && data != undefined) {
+      await fs.writeFileSync(filePath, data);
     }
+  }
+};
+
+async function readDirectory(dirPath) {
```
Comment (AtishayMsft) [inline tools/UnitTestApp/script.ts:36]: which function is to be used readDirectory or readDirectoryDeep?

File: tools/UnitTestApp/Sparkline/SparklineUT.test.tsx (hunk 1 above)
Comment (srmukher) [reply tools/UnitTestApp/Sparkline/SparklineUT.test.tsx]: updated

File: tools/UnitTestApp/TestUtility.ts (hunk 2 above)
Comment (srmukher) [reply tools/UnitTestApp/TestUtility.ts]: updated

File: tools/UnitTestApp/script.ts (hunk 3 above)
Comment (srmukher) [reply tools/UnitTestApp/script.ts]: updated

File: tools/UnitTestApp/script.ts (hunk 4 above)
Comment (srmukher) [reply tools/UnitTestApp/script.ts]: done

File: tools/UnitTestApp/script.ts (hunk 5 above)
Comment (srmukher) [reply tools/UnitTestApp/script.ts]: both for different purposes, for some files like Utility, only shallow updation is enough. But for base js modules, we need deep updation.

File: tools/UnitTestApp/TestSetupIndex.js [hunk 6]
```diff
@@ -111,9 +111,6 @@ const testSetupRewire = () => {
   // core.setOutput("ModifiedChart", chart.toString());
   const indexFilePath = `${npmRoot}/@fluentui/react-charting/lib-commonjs/index.js`;
   setupIndexFile(indexFilePath, null);
-  // const indexFileContents = require(indexFilePath);
-  // core.setOutput("ModifiedIndex", indexFileContents.toString());
 };
 
-testSetupRewire();
-module.exports=testSetupRewire;
+testSetupRewire();
```
Comment (AtishayMsft) [inline tools/UnitTestApp/TestSetupIndex.js:116]: why is index.js file in dist folder

File: tools/UnitTestApp/script.ts (hunk 5 above)
Comment (AtishayMsft) [reply tools/UnitTestApp/script.ts]: add comments for it if not done already.

File: tools/UnitTestApp/TestSetupIndex.js (hunk 6 above)
Comment (srmukher) [reply tools/UnitTestApp/TestSetupIndex.js:116]: the index.js file is generated in dist folder by default as a part of ncc build using vercel/ncc

Comment (AtishayMsft) [reply tools/UnitTestApp/TestSetupIndex.js:116]: then where is its content coming from if the file is autogenerated. And how to change the code in this fle.
I see that you have deleted the script.ts that had the code.

**Pull request closed without merging**